### DISCUSSION TITLE
composer require nativephp/tauri

### DISCUSSION THREAD

**Discussion by Silomalo** [NativePHP/laravel#237] [Q&A] [answered]
Hello @simonhamp , @DanielHudson, @ArondeParon and 
Marcel Pociot
I would like to use tauri to handle the project due to its small size in the final bundle. But I cannot see the installation process using tauri in the official documentation: https://nativephp.com/docs/1/getting-started/installation.
Kindly how can I achieve this?

Also is it safe for me to build an installer where its source code can not be stolen or reverse-engineered?
I have a pending SAAS project that I would like to create using Tauri how soon will native php be production ready, I have seen there has been a delay since August with little changes to the repos also YouTube tutorials have stopped. I understand you guys are busy but this looks promising and am looking forward to using it.  Can I start with what we have or much will change when it becomes production-ready?
Thank you.

**Comment by simonhamp** [comment, marked as answer]
@Silomalo

>I would like to use tauri to handle the project due to its small size in the final bundle.

The Tauri driver is not yet available, but it's coming! I am working on this currently. Look out for more updates here and on nativephp.com in the coming weeks.

>Also is it safe for me to build an installer where its source code can not be stolen or reverse-engineered?

I'm not quite sure what you mean by this. We don't currently have a way to secure the PHP code within a bundled application as there are many challenges to overcome there, besides the ones we currently have around the rest of NativePHP.

Note that all systems can be reverse-engineered eventually, the main challenge is around making it cost-prohibitive to do so.

If you value an app that you're intending to build in the $millions, then I would highly recommend not using NativePHP to build and deploy your solution at this stage.

>Can I start with what we have or much will change when it becomes production-ready?

We follow semantic versioning of all of NativePHP's repos.

But it's easiest to think of NativePHP in terms of being simply the main package that your Laravel application uses (this repo, `nativephp/laravel`), so this package itself is really the only thing you need to worry about.

(Although your application will directly require one of the NativePHP drivers, `nativephp/electron` or `nativephp/tauri`, these are simply thin, platform-specific wrappers around `nativephp/laravel`.)

It's currently on v0.4 which is a pre-production release. This is meant to give developers an opportunity to exercise the API and find bugs etc whilst also signalling that things are likely to change.

Although the API has been stable for some time and right now I don't envisage major changes, I cannot guarantee that. So if you choose to start building before this package hits v1, there will always be some risk that there will be changes that you will need to accommodate.

The other thing to note here is that it is our intention to make `nativephp/electron` and `nativephp/tauri` somewhat interchangeable. Now, while I'm almost certain that it won't be as simple as just changing the package name in your `composer.json`, the underlying APIs will be mostly from `nativephp/laravel` so the impact should be minimal.

That means you should be able to start building your proof of concept, testing the API and potentially even get an MVP into the hands of your customers using the currently-available Electron driver and then switch to the Tauri driver at a later stage, when that is available.
